worker/codalabworker/local_run/local_run_state.py

```diff
@@ -0,0 +1,277 @@
+
+            docker_dependency_path = os.path.join(docker_dependencies_path, dep['child_path'])
+
+            os.symlink(docker_dependency_path, child_path)
+            # These are turned into docker volume bindings like:
+            #   dependency_path:docker_dependency_path:ro
+            dependencies.append((dependency_path, docker_dependency_path))
+
+        # 3) Set up container
+        if run_state.resources['request_network']:
+            docker_network = self._run_manager.docker_network_external_name
+        else:
+            docker_network = self._run_manager.docker_network_internal_name
+
+        try:
+            cpuset, gpuset = self._run_manager.assign_cpu_and_gpu_sets(
+                run_state.resources['request_cpus'], run_state.resources['request_gpus'])
+        except:
+            run_state.info['failure_message'] = "Cannot assign enough resources"
+            return run_state._replace(stage=LocalRunStage.CLEANING, info=run_state.info)
+
+        # 4) Start container
+        container_id = self._run_manager.docker.start_container(
+            run_state.bundle_path, bundle_uuid, run_state.bundle['command'],
+            run_state.resources['docker_image'], docker_network, dependencies,
+            cpuset, gpuset, run_state.resources['request_memory']
+        )
+
+        digest = self._run_manager.docker.get_image_repo_digest(run_state.resources['docker_image'])
+
+        return run_state._replace(stage=LocalRunStage.RUNNING, run_status='Running',
+            container_id=container_id, docker_image=digest, has_contents=True, cpuset=cpuset, gpuset=gpuset)
+
+    def _transition_from_RUNNING(self, run_state):
+        def check_and_report_finished(run_state):
+            try:
+                finished, exitcode, failure_msg = self._run_manager.docker.check_finished(run_state.container_id)
+            except DockerException:
+                traceback.print_exc()
+                finished, exitcode, failure_msg = False, None, None
+            return dict(finished=finished, exitcode=exitcode, failure_message=failure_msg)
+
+        bundle_uuid = run_state.bundle['uuid']
+
+        new_info = check_and_report_finished(run_state)
+        run_state.info.update(new_info)
+        run_state = run_state._replace(info=run_state.info)
+
+        if run_state.is_killed and run_state.container_id is not None:
+            try:
+                self._run_manager.docker.kill_container(run_state.container_id)
+            except DockerException:
+                traceback.print_exc()
+            return run_state._replace(stage=LocalRunStage.CLEANING, container_id=None)
+        if run_state.info['finished']:
+            logger.debug('Finished run with UUID %s, exitcode %s, failure_message %s',
+                 bundle_uuid, run_state.info['exitcode'], run_state.info['failure_message'])
+            return run_state._replace(stage=LocalRunStage.CLEANING, run_status='Uploading results')
```

Comment:
this run status message is inaccurate

Reply:
It's a bit weird since the run status really aligns better with transition function for the the target stage (`_transition_from_CLEANING`) rather than this function.

Reply:
These transition functions prepare the run state for the next stage, so in this case the return RunState object will be FINISHED.